connector/connect/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -56,7 +56,7 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[Response]) exte
     try {
       processAsArrowBatches(request.getClientId, dataframe)
     } catch {
-      case e: Exception =>
+      case e: Throwable =>
```

Comment:
maybe `NonFatal`?

Reply:
Hm, it's a fallback so should better work in any case .. (?). I think we should remove this fallback anyway. I can revert this change for now too. (see https://github.com/apache/spark/pull/38468#discussion_r1013555606

Reply:
You are catching OutOfMemory and other virtual machine errors... You won't be able to recover from that.

Reply:
Yeah, let me revert this for now. We should remove this JSON fallback anyway.